### PR TITLE
EVG-6693 Remove base commit arg from patch-file command

### PR DESCRIPTION
**Pull request by bsamek**

### CODE REVIEW

File: operations/patch.go [hunk 1]
```diff
@@ -160,7 +156,18 @@ func PatchFile() cli.Command {
...
 			}
 
-			if _, err = params.validatePatchCommand(ctx, conf, ac, comm); err != nil {
+			ref, err := params.validatePatchCommand(ctx, conf, ac, comm)
+			if err != nil {
+				return err
+			}
+			var featureBranch string
+			if params.Ref == "" {
+				featureBranch = "HEAD"
```
Comment (ybrill) [inline operations/patch.go]: with a patch-file, is it not feasible that after generating the patch the user has checked out something else or even `cd`ed out of the repo?

Comment (bsamek) [reply operations/patch.go]: It's certainly conceivable. @john-m-liu @tychoish Any idea how people actually use this feature? I'd prefer to require that the user have the commit checked out, but I'm not sure what the actual use case is for patch-file.

Comment (bsamek) [reply operations/patch.go]: Actually, thinking about this again, I don't think this works. For `evergreen patch`, Evergreen does the diff, so we can do it against head. For a patch-file, we don't know what its base is.

**Comment by tychoish** [comment]
I'm not sure that people use patch file very much. I think I'd probably opt for a more conservative solution where you allow base to be settable and default to the same thing that normal patches commit. 

I think requiring people to have things checked out, is probably not good, and I don't feel great about such a fundamental change to the semantics of the command. 

Having said that I have no idea why we're doing this change now....

**Comment by bsamek** [comment]
I'm going to close this and think about another approach.

**Pull request closed without merging**